### PR TITLE
Create Error Component

### PR DESCRIPTION
**Pull request by sam-warren**
# Fixes or Implements [AB#8764](https://qslvic.visualstudio.com/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/8764)

* [X] Enhancement
* [ ] Bug
* [ ] Documentation

## Reviewer Guidelines

A reviewer should examine the PR if the Status above indicates the PR is **Ready**.
Once ready, the reviewer should read the information below as well as supporting information like the Azure ticket to ensure they have a good understanding of the request.

The reviewer should then:

1. Is the PR template complete and accurate?

    Confirm by reading all sections below.

1. Confirm the merge branch

    A normal feature/bug branch should merge into the Dev branch.  Exceptions for hotfix and sprint merges do occur and the target would then be Master.

1. Do the code commits have an appropriate comment and include linkage to the azure ticket by including a reference to "AB#nnnn"? 

1. Review the Code Changes

    1. Do the changes match the description?

        The code commits should reflect the documented description and nothing more.  If additional changes are included the reviewer should call these out and request that the developer document the changes in Azure and in the description or remove the changed code.

    1. useful tips

        * Any obvious errors or omissions?
        * Does the code match the style of other similar work?
        * Is the code easy to understand?
        * Is the code well documented?
        * DB Migrations are appropriate?

If the Answer to any of the questions above is No, then a request for change should be the result of the review.

## Description

Added an error card component that accepts Title, Description, Code and Show as props to pass information through.

## Testing

* [ ] Unit Tests Updated
* [ ] Functional Tests Updated
* [X] Not Required

### UI Changes

NO

### Browsers Tested

* [ ] Chrome
* [ ] Safari
* [ ] Edge
* [X] Firefox

### CODE REVIEW

File: Apps/WebClient/src/ClientApp/tsconfig.json [hunk 1]
```diff
@@ -10,6 +10,7 @@
     "allowSyntheticDefaultImports": true,
     "sourceMap": true,
     "isolatedModules": true,
+    "strictPropertyInitialization": false,
```
Comment (FuriousLlama) [inline Apps/WebClient/src/ClientApp/tsconfig.json:13]: I think we want this to be true, that way the type checker makes sure all variables are properly initialized

File: Apps/WebClient/src/ClientApp/src/components/timeline/errorCard.vue [hunk 2]
```diff
@@ -0,0 +1,29 @@
+<template>
```
Comment (FuriousLlama) [inline Apps/WebClient/src/ClientApp/src/components/timeline/errorCard.vue:1]: this looks useful outside the timeline. How about moving it to the components folder (parent folder from this).

Comment (sam-warren) [reply Apps/WebClient/src/ClientApp/src/components/timeline/errorCard.vue:1]: Sure!

File: Apps/WebClient/src/ClientApp/tsconfig.json (hunk 1 above)
Comment (sam-warren) [reply Apps/WebClient/src/ClientApp/tsconfig.json:13]: Kk

**Comment by leobcgov** [comment]
Thanks @sam-warren , do you have a screenshot available that you may have used to test this new vue?

File: Apps/WebClient/src/ClientApp/src/components/timeline/errorCard.vue [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+@Component
+export default class ErrorCardComponent extends Vue {
+    @Prop() title: string;
+    @Prop() description: string;
+    @Prop() code: string;
+    @Prop() show: boolean;
+
+    private getCodeText(): string {
+        if (!this.code) {
```
Comment (leobcgov) [inline Apps/WebClient/src/ClientApp/src/components/timeline/errorCard.vue:22]: how about a shorter/1-line version of this as follows?
`return !this.code ? "" : this.code + ": ";`

Comment (FuriousLlama) [reply Apps/WebClient/src/ClientApp/src/components/timeline/errorCard.vue:22]: i would make it the other way around ```return this.code ? this.code + ": ": "" ;

Comment (sam-warren) [reply Apps/WebClient/src/ClientApp/src/components/timeline/errorCard.vue:22]: Will do.

File: Apps/WebClient/src/ClientApp/src/components/timeline/errorCard.vue (hunk 2 above)
Comment (sam-warren) [reply Apps/WebClient/src/ClientApp/src/components/timeline/errorCard.vue:1]: Done

File: Apps/WebClient/src/ClientApp/src/components/timeline/errorCard.vue (hunk 3 above)
Comment (leobcgov) [reply Apps/WebClient/src/ClientApp/src/components/timeline/errorCard.vue]: > i would make it the other way around `return this.code ? this.code + ": ": "" ;`
Way better with this :)

File: Apps/WebClient/src/ClientApp/src/components/errorCard.vue [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+@Component
+export default class ErrorCardComponent extends Vue {
+    @Prop() title: string = "";
+    @Prop() description: string = "";
+    @Prop() code: string = "";
+    @Prop() show: boolean = false;
+
+    private getCodeText(): string {
+        return !this.code ? "" : this.code + ": ";
```
Comment (leobcgov) [inline Apps/WebClient/src/ClientApp/src/components/errorCard.vue:22]: per @FuriousLlama , should we flip this code as follow, as it looks more natural :-)
`return this.code ? this.code + ": " : "";`

**Pull request merged**